### ISSUE TITLE
Compound values are lost when rendering due to not-matching registration in DB

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#3282]
The following entity in DB:

```
{
	"_id" : {
		"id" : "E10",
		"type" : "T",
		"servicePath" : "/"
	},
	"attrNames" : [
		"A",
		"B",
		"C"
	],
	"attrs" : {
		"A" : {
			"type" : "Float",
			"creDate" : 1535729080,
			"modDate" : 1535729080,
			"value" : 26.5,
			"md" : {
				"MD1" : {
					"type" : "Float",
					"value" : 0.8
				},
				"MD2" : {
					"type" : "Float",
					"value" : {
						"z" : 1,
						"w" : 2
					}
				},
				"MD3" : {
					"type" : "Float",
					"value" : [
						"B",
						5,
						false
					]
				}
			},
			"mdNames" : [
				"MD1",
				"MD2",
				"MD3"
			]
		},
		"B" : {
			"type" : "Text",
			"creDate" : 1535729080,
			"modDate" : 1535729080,
			"value" : {
				"x" : 1,
				"y" : 2
			},
			"md" : {
				"accuracy" : {
					"type" : "Float",
					"value" : 0.8
				}
			},
			"mdNames" : [
				"accuracy"
			]
		},
		"C" : {
			"type" : "Boolean",
			"creDate" : 1535729080,
			"modDate" : 1535729080,
			"value" : [
				"A",
				2,
				true
			],
			"md" : {
				"accuracy" : {
					"type" : "Float",
					"value" : 0.8
				}
			},
			"mdNames" : [
				"accuracy"
			]
		}
	},
	"creDate" : 1535729080,
	"modDate" : 1535729080,
	"lastCorrelator" : "fa5cf03a-ad31-11e8-8dad-000c29173617"
}
```

The following registration in DB:

```
> db.registrations.find().pretty()
{
	"_id" : ObjectId("5b69f1df96d022bca0a545ef"),
	"expiration" : NumberLong("9223372036854775807"),
	"servicePath" : "/fermin",
	"contextRegistration" : [
		{
			"entities" : [
				{
					"id" : "E",
					"type" : "T"
				}
			],
			"attrs" : [
				{
					"name" : "C1",
					"type" : "",
					"isDomain" : "false"
				},
				{
					"name" : "C2",
					"type" : "",
					"isDomain" : "false"
				}
			],
			"providingApplication" : "http://localhost:1027/v1"
		}
	],
	"format" : "JSON"
}
```

Note the registration **does not match** the entity.

The following GET operations fail as they are not rendering the compound values (in the keyValues case, note that even the rendering is illegal JSON):

```
$ curl 'localhost:1026/v2/entities?q=A==26.5&options=keyValues'

[{"id":"E10","type":"T","A":26.5,"B":,"C":}]
```

```
$ curl 'localhost:1026/v2/entities?q=A==26.5' | python -mjson.tool
[
    {
        "A": {
            "metadata": {
                "MD1": {
                    "type": "Float",
                    "value": 0.8
                },
                "MD2": {
                    "type": "Float",
                    "value": {
                        "w": 2,
                        "z": 1
                    }
                },
                "MD3": {
                    "type": "Float",
                    "value": [
                        "B",
                        5,
                        false
                    ]
                }
            },
            "type": "Float",
            "value": 26.5
        },
        "B": {
            "metadata": {
                "accuracy": {
                    "type": "Float",
                    "value": 0.8
                }
            },
            "type": "Text",
            "value": ""
        },
        "C": {
            "metadata": {
                "accuracy": {
                    "type": "Float",
                    "value": 0.8
                }
            },
            "type": "Boolean",
            "value": ""
        },
        "id": "E10",
        "type": "T"
    }
]
```

If we use GET /v2/entities/E10 instead of GET /v2/entities it works:

```
$ curl 'localhost:1026/v2/entities/E10?options=keyValues&q=A==26.5'

{"id":"E10","type":"T","A":26.5,"B":{"x":1,"y":2},"C":["A",2,true]}
```

```
$ curl 'localhost:1026/v2/entities?q=A==26.5' | python -mjson.tool
[
    {
        "A": {
            "metadata": {
                "MD1": {
                    "type": "Float",
                    "value": 0.8
                },
                "MD2": {
                    "type": "Float",
                    "value": {
                        "w": 2,
                        "z": 1
                    }
                },
                "MD3": {
                    "type": "Float",
                    "value": [
                        "B",
                        5,
                        false
                    ]
                }
            },
            "type": "Float",
            "value": 26.5
        },
        "B": {
            "metadata": {
                "accuracy": {
                    "type": "Float",
                    "value": 0.8
                }
            },
            "type": "Text",
            "value": ""
        },
        "C": {
            "metadata": {
                "accuracy": {
                    "type": "Float",
                    "value": 0.8
                }
            },
            "type": "Boolean",
            "value": ""
        },
        "id": "E10",
        "type": "T"
    }
]
```

If we remove the registration CB works as expected

**Comment by fgalan** [comment]
Debugging this issue I have found in postQueryContext() function in postQueryContext.cpp:

```
          if (contextElementResponseP == NULL)
          {
            contextElementResponseP = new ContextElementResponse(eP, aP);
            localQcrsP->contextElementResponseVector.push_back(contextElementResponseP);
          }
          else
          {
            contextElementResponseP->contextElement.contextAttributeVector.push_back(new ContextAttribute(aP)); <---- (1)
          }
```

qcrsP.contextElementResponseVector.vec[0].contextElement.contextAttributeVector.vec[1] and qcrsP.contextElementResponseVector.vec[0].contextElement.contextAttributeVector.vec[2] correspond to attributes B and C in E10 (E10 is qcrsP.contextElementResponseVector.vec[0]). The compoundValueP is ok in both until (1) is executed. When (1) is executed, compoundValueP is set to 0x0 (i.e. NULL). That is probably  causing the compound is not rendered at the end.

**Comment by fgalan** [comment]
Solved by PR #3380
